spec/lib/method_hooks_spec.rb

```diff
@@ -81,4 +94,13 @@ def save
 
   end
 
+  it 'does not override existing .method_added hook' do
+    Base.class_eval do
+      def some_method
+      end
+    end
+
+    expect(Base._methods).to include(:some_method)
+  end
+
```

Comment:
I would put the module definition in this test, since it's not needed in the other tests. Here's how I would do it:

```ruby
it 'does not override existing .method_added hook' do
  module AnotherModuleWIthMethodAddedHook
    def method_added(method_name)
      _methods << method_name
      super
    end

    def _methods
      @_methods ||= []
    end
  end

  Base.extend(AnotherModuleWIthMethodAddedHook)

  Base.class_eval do
    def some_method
    end
  end

  expect(Base._methods).to include(:some_method)
end
```

What do you think?